pandas/core/arrays/timedeltas.py

```diff
@@ -381,14 +381,12 @@ def sum(
         nv.validate_sum(
             (), dict(dtype=dtype, out=out, keepdims=keepdims, initial=initial)
         )
-        if not self.size and (self.ndim == 1 or axis is None):
-            return NaT
 
         result = nanops.nansum(
-            self._data, axis=axis, skipna=skipna, min_count=min_count
+            self._ndarray, axis=axis, skipna=skipna, min_count=min_count
         )
-        if is_scalar(result):
-            return Timedelta(result)
+        if axis is None or self.ndim == 1:
```

Comment:
is the ndim >1 tested at all? why is this needed

Reply:
> is the ndim >1 tested at all?

yes, there are dedicated tests for this.

> why is this needed

1) completeness/correctness
2) fixing the reduction bugs is going to involve doing more things blockwise, not going through frame.values